packages/shared/routes/dashboard/wallet/views/Receive.svelte

```diff
@@ -18,24 +18,23 @@
     export let onGenerateAddress = (accountId: AccountIdentifier): void => {}
 
     const liveAccounts = getContext<Readable<WalletAccount[]>>('liveAccounts')
-    const currentAccount = getContext<Readable<WalletAccount>>('selectedAccount')
 
-    let selectedAccount = $currentAccount || $liveAccounts[0]
+    let selectedSendAccount = $selectedAccount
 
     const handleDropdownSelect = (item: WalletAccount): void => {
-        selectedAccount = item
+        selectedSendAccount = item
     }
     const generateNewAddress = (): void => {
-        onGenerateAddress(selectedAccount.id)
+        onGenerateAddress(selectedSendAccount.id)
     }
     const handleCloseClick = (): void => {
         accountRoute.set(AccountRoutes.Init)
     }
 </script>
 
-<div class="w-full h-full flex flex-col justify-between {!$currentAccount ? 'p-8' : ''}">
+<div class="w-full h-full flex flex-col justify-between {!$selectedAccount ? 'p-8' : ''}">
     <div class="w-full h-full space-y-6 flex flex-auto flex-col flex-shrink-0">
-        {#if !$currentAccount}
+        {#if !$selectedAccount}
```

Comment:
Does this scenario occur? I couldn't recreate it, since there always has to be one account. We should check this, cuz we might just remove a lot of code here :heart_eyes:

Reply:
Maybe also by defaulting on the 0th index in the derived store if the account ID cannot be found.

Reply:
I'll work on these changes on a separate PR, im working on it  in branch `feat/cleanup`

Reply:
Top!